### ISSUE TITLE
Error when accessing TrustyAI service through ODH dashboard

### ISSUE THREAD

**Issue by adolfo-ab** [trustyai-python/trustyai-explainability#554]
**Description**
After deploying a model in ODH, sending training data and scheduling a fairness metric request, when trying to access the Model Bias screen through the ODH dashboard, an error screen appears stating that an error has been encountered when accessing the TrustyAI service. Reproduced while following [this demo
](https://github.com/trustyai-explainability/odh-trustyai-demos/tree/main/2-BiasMonitoring#schedule-a-fairness-metric-request) using ODH UI.

**Preconditions**
- OpenShift cluster with ODH (used OCP 4.14.16 and ODH 2.11)
- Data Science Project, with the serving runtime, model server and models (one model server for each model) described in the demo above (deployed through ODH UI instead of through CLI)
- TrustyAI enabled for the project.

**Steps to reproduce**
- Send training data for both models, as described in the demo above.
- Label data fields, as described in the demo above.
- Define the following bias metric for both models through the UI:
                 "protectedAttribute": "Is Male-Identifying?",
                 "privilegedAttribute": 1.0,
                 "unprivilegedAttribute": 0.0,
                 "outcomeName": "Will Default?",
                 "favorableOutcome": 0,
                 "batchSize": 5000
- Schedule a fairness Metric request, Identity metric request and send real-world data, as described in the demo above.
- Try to access the Model Serving > Model Bias screen in the ODH dashboard.

**Expected result**
The screen is shown, and the defined metrics can be properly visualized.

**Actual result**
An error screen appears with the following message:

> TrustyAI Error  
We encountered an error accessing the TrustyAI service:  
Cannot read properties of undefined (reading 'value')

When in this error screen, if you click on the "Configure" button, a loading screen appears but the content never actually loads (see attached images).

Even if a new model servers  and models are created, when you try to access their corresponding Model Bias screen in the dashboard, the same error appears.

Error logs:
[trustyai-service-oauth-proxy.txt](https://github.com/trustyai-explainability/trustyai-service-operator/files/15151070/trustyai-service-oauth-proxy.txt)
[trustyai-service.txt](https://github.com/trustyai-explainability/trustyai-service-operator/files/15151071/trustyai-service.txt)
[trustyai-service-operator-controller-manager.txt](https://github.com/trustyai-explainability/trustyai-service-operator/files/15151072/trustyai-service-operator-controller-manager.txt)

Screenshots:
![error-screen](https://github.com/trustyai-explainability/trustyai-service-operator/assets/83642232/14b5063d-ba02-4758-9bb5-45819f3aeb9c)
![loading](https://github.com/trustyai-explainability/trustyai-service-operator/assets/83642232/567153d5-69c9-4b2e-aef8-eac5fe98a447)

**Comment by ruivieira** [comment]
Thanks for the report @adolfo-ab.

This issue involves the ODH Dashboard too.
When in the "metrics" Dashboard page, a request equivalent to

```
curl 'https://odh-dashboard-opendatahub.<redacted>/api/service/trustyai/model-namespace/trustyai-service/metrics/all/'
```

is performed. This requests all registered metrics in the TrustyAI service.
Typically, if only fairness/bias (eg SPD/DIR) metrics are registered, the response will be

```json
{
    "requests": [
        {
            "id": "dc171b99-0b77-4a92-b8be-e1c855d1749a",
            "request": {
                "@type": "GroupMetricRequest",
                "modelId": "demo-loan-nn-onnx-alpha",
                "requestName": "SPD",
                "metricName": "SPD",
                "batchSize": 5000,
                "protectedAttribute": "Is Male-Identifying?",
                "outcomeName": "Will Default?",
                "privilegedAttribute": {
                    "type": "DOUBLE",
                    "value": 1
                },
                "unprivilegedAttribute": {
                    "type": "DOUBLE",
                    "value": 0
                },
                "favorableOutcome": {
                    "type": "INT64",
                    "value": 0
                },
                "thresholdDelta": 0.1
            }
        },
        {
            "id": "a149e32a-8df1-47d7-a227-7c67cadef334",
            "request": {
                "@type": "GroupMetricRequest",
                "modelId": "demo-loan-nn-onnx-alpha",
                "requestName": "DIR",
                "metricName": "DIR",
                "batchSize": 5000,
                "protectedAttribute": "Is Male-Identifying?",
                "outcomeName": "Will Default?",
                "privilegedAttribute": {
                    "type": "DOUBLE",
                    "value": 1
                },
                "unprivilegedAttribute": {
                    "type": "DOUBLE",
                    "value": 0
                },
                "favorableOutcome": {
                    "type": "INT64",
                    "value": 0
                },
                "thresholdDelta": 0.2
            }
        }
    ]
}
```

And the dashboard will populate the page from that (importantly, fetching `request.favorableOutcome`.
However, if there's an identity metric registered, the response will be similar to

```json
{
    "requests": [
        {
            "id": "13a4bb60-22f8-4289-97cc-7335a4a71b59",
            "request": {
                "@type": "IdentityMetricRequest",
                "modelId": "...",
                "requestName": null,
                "metricName": "IDENTITY",
                "batchSize": 250,
                "columnName": "Is Male-Identifying?",
                "lowerThreshold": 0.0,
                "upperThreshold": 0.0
            }
        },
        {
            "id": "dc171b99-0b77-4a92-b8be-e1c855d1749a",
            "request": {
                "@type": "GroupMetricRequest",
                "modelId": "demo-loan-nn-onnx-alpha",
                "requestName": "SPD",
                "metricName": "SPD",
                "batchSize": 5000,
                "protectedAttribute": "Is Male-Identifying?",
                "outcomeName": "Will Default?",
                "privilegedAttribute": {
                    "type": "DOUBLE",
                    "value": 1
                },
                "unprivilegedAttribute": {
                    "type": "DOUBLE",
                    "value": 0
                },
                "favorableOutcome": {
                    "type": "INT64",
                    "value": 0
                },
                "thresholdDelta": 0.1
            }
        },
        {
            "id": "a149e32a-8df1-47d7-a227-7c67cadef334",
            "request": {
                "@type": "GroupMetricRequest",
                "modelId": "demo-loan-nn-onnx-alpha",
                "requestName": "DIR",
                "metricName": "DIR",
                "batchSize": 5000,
                "protectedAttribute": "Is Male-Identifying?",
                "outcomeName": "Will Default?",
                "privilegedAttribute": {
                    "type": "DOUBLE",
                    "value": 1
                },
                "unprivilegedAttribute": {
                    "type": "DOUBLE",
                    "value": 0
                },
                "favorableOutcome": {
                    "type": "INT64",
                    "value": 0
                },
                "thresholdDelta": 0.2
            }
        }
    ]
}
```

Making the page fail since there is not `request.favorableOutcome` in the identity metrics.
